File: Axiom/Internal/Ast/MemberExpression.cs
namespace Axiom.Internal.Ast
{
    internal class MemberExpression : Expression
    {
        internal enum AccessType
        {
            ListAccess, Property, Call
        }

        public MemberExpression(AccessType type, Expression obj, Expression member)
        {
            Object = obj;
            Member = member;
            Type = type;
        }

        public override void Accept(IVisitor visitor)
        {
            visitor.Visit(this);
        }

        public Expression Object { get; private set; }
        public Expression Member { get; private set; }
        public AccessType Type { get; private set; }
    }
}

File: src/main/java/com/moybl/axiom/ast/MemberExpression.java
package com.moybl.axiom.ast;

import com.moybl.axiom.Position;

public class MemberExpression extends Expression {

	public enum Kind {
		LIST, PROPERTY
	}

	private Kind kind;
	private Expression object;
	private Expression member;

	public MemberExpression(Position position, Kind kind, Expression object, Expression member) {
		super(position);
		this.kind = kind;
		this.object = object;
		this.member = member;
	}

	public void accept(Visitor visitor) {
		visitor.visit(this);
	}

	public Kind getKind() {
		return kind;
	}

	public Expression getObject() {
		return object;
	}

	public Expression getMember() {
		return member;
	}

}
